app/views/dev_mode.py

```diff
@@ -91,19 +92,12 @@ def dev_flush_mode_get():
 
 def extract_eq_id_and_form_type(schema_name):
     logger.debug("extracting eq_id and form type from schema name", schema_name=schema_name)
-    if "_" in schema_name:
-        split_schema_name = schema_name.split("_", 1)
-        if len(split_schema_name) != 2:
-            raise ValueError("Schema file name incorrect %", schema_name)
-        eq_id = split_schema_name[0]
-        split_rest_of_name = split_schema_name[1].split(".", 1)
-        if len(split_rest_of_name) != 2:
-            raise ValueError("Schema file name incorrect %", schema_name)
-        form_type = split_rest_of_name[0]
+    match = re.match(r"^([a-z0-9]+)_(\w+)\.json$", schema_name)
+    if match:
+        eq_id = match.group(1)
+        form_type = match.group(2)
     else:
-        # No form type associated with
-        eq_id = schema_name.split(".", 1)[0]
-        form_type = "-1"
+        raise ValueError('Invalid schema format')
```

Comment:
Last comment, previously this output the filename of the invalid file on error, it would be good to keep that:

`raise ValueError("Schema filename incorrect format: %", schema_name)`